clang/test/CIR/CodeGen/same-name-struct.c

```diff
@@ -0,0 +1,34 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s
+
+typedef struct demo{
```

Comment:
When I try to build the testcase without your PR, the crash doesn't seem much related with fixing the unique names, what am I missing?